aws-rds-dbcluster/src/main/java/software/amazon/rds/dbcluster/BaseHandlerStd.java

```diff
@@ -1,163 +1,303 @@
+    protected boolean isDBClusterDeleted(
+            final ProxyClient<RdsClient> proxyClient,
+            final ResourceModel model
+    ) {
+        try {
+            fetchDBCluster(proxyClient, model);
+        } catch (DbClusterNotFoundException e) {
+            return true;
+        }
+        return false;
     }
 
-    // Modify or Post Create
-    protected ProgressEvent<ResourceModel, CallbackContext> modifyDBCluster(final AmazonWebServicesClientProxy proxy,
-                                                                            final ProxyClient<RdsClient> proxyClient,
-                                                                            final ProgressEvent<ResourceModel, CallbackContext> progress,
-                                                                            final CloudwatchLogsExportConfiguration config) {
-        if (progress.getCallbackContext().isModified()) return progress;
-        return proxy.initiate("rds::modify-dbcluster", proxyClient, progress.getResourceModel(), progress.getCallbackContext())
-            .translateToServiceRequest((modelRequest) -> modifyDbClusterRequest(modelRequest, config))
-            .backoffDelay(BACKOFF_STRATEGY)
-            .makeServiceCall((dbClusterModifyRequest, proxyInvocation) -> proxyInvocation.injectCredentialsAndInvokeV2(dbClusterModifyRequest, proxyInvocation.client()::modifyDBCluster))
-            .done((modifyDbClusterRequest, modifyDbClusterResponse, proxyInvocation, resourceModel, callbackContext) ->  {
-                callbackContext.setModified(true);
-                return ProgressEvent.defaultInProgressHandler(callbackContext, PAUSE_TIME_SECONDS, resourceModel);
-            });
+    protected ProgressEvent<ResourceModel, CallbackContext> addAssociatedRoles(
+            final AmazonWebServicesClientProxy proxy,
+            final ProxyClient<RdsClient> proxyClient,
+            final ProgressEvent<ResourceModel, CallbackContext> progress,
+            final List<DBClusterRole> roles
+    ) {
+        final ResourceModel model = progress.getResourceModel();
+        final CallbackContext callbackContext = progress.getCallbackContext();
+
+        for (final DBClusterRole dbClusterRole : Optional.ofNullable(roles).orElse(Collections.emptyList())) {
+            final ProgressEvent<ResourceModel, CallbackContext> progressEvent = proxy.initiate("rds::add-roles-to-dbcluster", proxyClient, model, callbackContext)
+                    .translateToServiceRequest(modelRequest -> addRoleToDbClusterRequest(
+                            modelRequest.getDBClusterIdentifier(),
+                            dbClusterRole.getRoleArn(),
+                            dbClusterRole.getFeatureName()
+                    ))
+                    .makeServiceCall((modelRequest, proxyInvocation) -> proxyInvocation.injectCredentialsAndInvokeV2(
+                            modelRequest,
+                            proxyInvocation.client()::addRoleToDBCluster
```

Comment:
`AddRoleToDBCluster` can throw  `DBClusterRoleAlreadyExists`. So update handler can stuck in adding already exist role (ie: added manually before by customer Or duplicated role between create stack and Update stack). Is this behavior expected or should we handle it i.e: ignore this exception in ErrorRuleSet?

Reply:
Added this exception in a separate ruleset `ADD_ASSOC_ROLES_ERROR_RULE_SET`